### PR TITLE
[Address32][2/3] Replace script with script function

### PR DESCRIPTION
**Pull request by areshand**
## Motivation

Latest move is required to use 32 byte address. 
To use the latest move, need to release DPN and all the call-sites for executing a script now requires a script function. 
The intention of making this migration to make the txn script calling more explicit compared to running a script code directly. 

**Currently, scriptfunction writeset is not supported and two tests failed as executed_as is not supported after switching to script function. Created an issue as an follow up https://github.com/aptos-labs/internal-ops/issues/114**

### Have you read the [Contributing Guidelines on pull requests](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md#pull-requests)?

yes

## Test Plan

(Share your test plan here. If you changed code, please provide us with clear instructions for verifying that your changes work.)
cargo test
cargo x test
cargo xfmt && cargo xclippy --all-targets

### CODE REVIEW

File: aptos-move/e2e-testsuite/src/tests/transaction_builder.rs [hunk 1]
```diff
@@ -48,11 +47,7 @@ fn test_rotate_authentication_key_with_nonce_admin() {
 
         let account = test_env.dr_account;
         let txn = account
-            .transaction()
-            .write_set(WriteSetPayload::Script {
-                script: encode_rotate_authentication_key_with_nonce_admin_script(0, new_key_hash.clone()),
-                execute_as: *new_account.address(),
```
Comment (areshand) [inline aptos-move/e2e-testsuite/src/tests/transaction_builder.rs:54]: hi @zekun000  previous script allow passing an **_execute_as_**. however, no sure how we could pass execute_as for the script function. This tests failed.

File: aptos-move/e2e-testsuite/src/tests/validator_set_management.rs [hunk 2]
```diff
@@ -319,17 +319,13 @@ fn validator_set_operator_set_key_reconfigure() {
...
+        let admin_script_payload = encode_set_validator_operator_with_nonce_admin_script_function(
             0,
             b"operator_1".to_vec(),
             *operator_account_1.address(),
         );
         let txn = aptos_root_account
-            .transaction()
-            .write_set(WriteSetPayload::Script {
-                script: admin_script,
-                execute_as: *validator_account.address(),
```
Comment (areshand) [inline aptos-move/e2e-testsuite/src/tests/validator_set_management.rs:331]: hi @zekun000  this also have this execute_as. And I am not sure how to pass this in the new script function.

**Comment by areshand** [comment]
\canary

File: aptos-move/e2e-testsuite/src/tests/transaction_fuzzer.rs [hunk 3]
```diff
@@ -70,24 +70,24 @@ proptest! {
...
-            let script = txn.encode();
+            let payload = txn.encode();
             let (account, account_sequence_number) = accounts.get_mut(i % num_accounts).unwrap();
-            let script_is_rotate = LegacyStdlibScript::try_from(script.code()).map(|script|
+            /*let script_is_rotate = LegacyStdlibScript::try_from(script.code()).map(|script|
                 script == LegacyStdlibScript::RotateAuthenticationKey ||
                 script == LegacyStdlibScript::RotateAuthenticationKeyWithNonce ||
                 script == LegacyStdlibScript::RotateAuthenticationKeyWithRecoveryAddress
-            ).unwrap_or(false);
+            ).unwrap_or(false); */
```
Comment (davidiw) [inline aptos-move/e2e-testsuite/src/tests/transaction_fuzzer.rs:79]: delete?

File: aptos-move/e2e-testsuite/src/tests/transaction_fuzzer.rs [hunk 4]
```diff
@@ -70,24 +70,24 @@ proptest! {
...
                 .sequence_number(*account_sequence_number)
                 .sign());
                 prop_assert!(!output.status().is_discarded());
                 // Don't apply key rotation transactions since that will bork future txns
-                if !script_is_rotate {
+                /*if !script_is_rotate {
                     executor.apply_write_set(output.write_set());
                     *account_sequence_number += 1;
-                }
+                }*/
```
Comment (davidiw) [inline aptos-move/e2e-testsuite/src/tests/transaction_fuzzer.rs:90]: delete? or why is this commented out?

**Review by davidiw — COMMENTED** [review]
amazing, overall looks good except we should either annotate why we've commented out code / ignored tests or we should delete.

File: aptos-move/e2e-testsuite/src/tests/transaction_fuzzer.rs (hunk 3 above)
Comment (areshand) [reply aptos-move/e2e-testsuite/src/tests/transaction_fuzzer.rs:79]: This was done to pass the compile error. 
I delete this test completely now

File: aptos-move/e2e-testsuite/src/tests/transaction_fuzzer.rs (hunk 4 above)
Comment (areshand) [reply aptos-move/e2e-testsuite/src/tests/transaction_fuzzer.rs:90]: deleted the entire test now

**Review by zekun000 — APPROVED** [review]

**Pull request closed without merging**